src/OpenLoco/src/GameCommands/CreateIndustry.cpp

```diff
@@ -0,0 +1,767 @@
+        {
+            const auto tilePos = World::toTileSpace(pos + offset.pos);
+            if (!World::validCoords(tilePos))
+            {
+                return FAILURE;
+            }
+
+            if ((flags & Flags::apply) && !(flags & Flags::flag_6))
+            {
+                World::TileManager::removeAllWallsOnTileBelow(tilePos, highestBaseZ + clearHeight / World::kSmallZStep);
+            }
+
+            {
+                auto tile = World::TileManager::get(tilePos);
+                auto* surface = tile.surface();
+
+                auto clearFunc = [tilePos, flags, &totalCost](World::TileElement& el) {
+                    return tileClearFunction(el, World::toWorldSpace(tilePos), flags, totalCost);
+                };
+
+                // Perform clearance checks
+                if (indObj->hasFlags(IndustryObjectFlags::builtOnWater))
+                {
+                    if (surface->water() * World::kMicroToSmallZStep != highestBaseZ)
+                    {
+                        setErrorText(StringIds::can_only_be_built_on_water);
+                        return FAILURE;
+                    }
+
+                    if (surface->hasType6Flag())
+                    {
+                        setErrorText(StringIds::water_channel_currently_needed_by_ships);
+                        return FAILURE;
+                    }
+                    World::QuarterTile qt(0xF, 0xF);
+                    if (!World::TileClearance::applyClearAtStandardHeight(World::toWorldSpace(tilePos), highestBaseZ, clearZ, qt, clearFunc))
+                    {
+                        return FAILURE;
+                    }
+                }
+                else
+                {
+                    if (surface->water())
+                    {
+                        setErrorText(StringIds::cant_build_this_underwater);
+                        return FAILURE;
+                    }
+                    World::QuarterTile qt(0xF, 0xF);
+                    if (!World::TileClearance::applyClearAtStandardHeight(World::toWorldSpace(tilePos), surface->baseZ(), clearZ, qt, clearFunc))
+                    {
+                        return FAILURE;
+                    }
+                    // TODO: This is dangerous pointer might be invalid?
+                    if (surface->slope() || surface->baseZ() != highestBaseZ)
+                    {
+                        const auto* landObj = ObjectManager::get<LandObject>(surface->terrain());
+                        totalCost += Economy::getInflationAdjustedCost(landObj->costFactor, landObj->costIndex, 10);
+                    }
+                }
+            }
+
+            // Perform additional validation ?for ghosts? and flatten surfaces
+            if (!(flags & Flags::flag_6) && !indObj->hasFlags(IndustryObjectFlags::builtOnWater))
+            {
+                auto tile = World::TileManager::get(tilePos);
+                auto* surface = tile.surface();
```

Comment:
This shadows the previous surface variable, quite curious to how no warning is emitted.

Reply:
I purposely added extra {} to make this not shadow.